pkg/controllermanager/deployer/deployer.go

```diff
@@ -415,50 +417,64 @@ func (deployer *Deployer) populateBasesAndLocalizations(sub *appsapi.Subscriptio
 }
 
 func (deployer *Deployer) syncBase(sub *appsapi.Subscription, baseTemplate *appsapi.Base) (*appsapi.Base, error) {
-	base, err := deployer.clusternetClient.AppsV1alpha1().Bases(baseTemplate.Namespace).Create(context.TODO(),
-		baseTemplate, metav1.CreateOptions{})
-	if err == nil {
-		msg := fmt.Sprintf("Base %s is created successfully", klog.KObj(baseTemplate))
-		klog.V(4).Info(msg)
-		deployer.recorder.Event(sub, corev1.EventTypeNormal, "BaseCreated", msg)
-		return base, nil
-	}
-
-	if !apierrors.IsAlreadyExists(err) {
-		return nil, err
-	}
+	labelSelector := labels.SelectorFromSet(labels.Set{
+		known.ConfigKindLabel:      subscriptionKind.Kind,
+		known.ConfigNameLabel:      sub.Name,
+		known.ConfigNamespaceLabel: sub.Namespace,
+		known.ConfigUIDLabel:       string(sub.UID),
+	})
 
-	// update it
-	base, err = deployer.baseLister.Bases(baseTemplate.Namespace).Get(baseTemplate.Name)
+	curBases, err := deployer.clusternetClient.AppsV1alpha1().Bases(baseTemplate.Namespace).List(context.TODO(),
```

Comment:
why not getting bases from informer listers

Reply:
Also why not directly getting the base with the exact name with `baseTemplate.Name`? If the base object does not exist, then we could create with new name template.

Reply:
I didn't notice that there's a lister for base in the deployer object before.


If get base with expect name with notFound error, maybe an old name format base have been exist in cluster.
it means that there are actually for scenarios: 
1. no one base
2. have old format name base
3. have new format name base
4. more then one base

with 2 and 3, update it with same code logic, so list all base is a simple and straightforward way.